content/blog/pointers-and-references-cpp/index.md

```diff
@@ -116,30 +116,30 @@ The neat part of C++ is that you can control whether something is passed by refe
 	
 		int num1 = 0;
 		int num2 = 0;
-		cout << "Original values: " << num << endl;
+		cout << "Original value: " << num1 << endl;
 
 		// call passByReference()
 		passByReference(num1);
 
-		cout << "After passing by reference: " << num << endl;
+		cout << "After passing by reference: " << num1 << endl;
 		
 		// call passByValue()
 		passByValue(num2);
 
-		cout << "After passing by value: " << num << endl;
+		cout << "After passing by value: " << num2 << endl;
 		
 	}
 ```
 
 The output of this looks like...
 
 ```
-	Original values: 0
+	Original value: 0
 	After passing by reference: 2
 	After passing by value: 0
 ```
 
-As you can see, when passed by reference the local value is changed but when it is passed by value it is not even though they both perform the same operation. In languages outside of C++ — like Java for example — doesn't give you control of whether you pass by value or reference; Some types of objects or variables are passed by reference and some aren't. For example, Java collections are passed by reference, but all other variables and objects are passed by value.
+As you can see, when passed by reference the local value is changed but when it is passed by value it is not even though they both perform the same operation. In languages outside of C++ — like Java for example — doesn't give you control of whether you pass by value or reference; Java always passes by value, though it can pass addresses as values, making it seem like a pass by reference. For example, Java collections end up passing a reference by value, but all other variables are passed by value.
```

Comment:
Java collections seem to be passed by reference since they are a mutable object and you can use functions to change the object such as by using the `add` function. A big complaint that I have is that you make the statement "all other variables are passed by value." Yes, all variables are passed by value, but here is the distinction:

1. Java primitives are saved on the stack, these values are passed by value, a copy is passed in
2. Java objects have an address to the heap, where the object is stored, saved on the stack, where the address from the stack is passed by value. If you assign a new object to the object reference you have, it will only change in the scope of the function since the reference on the stack is copied to another place on the stack and then overwritten.

This is why primitives are copied and objects are not. This is a big distinction to think about and reword.